python/cuspatial/cuspatial/tests/binpreds/test_intersects_only_binpreds.py

```diff
@@ -0,0 +1,1017 @@
+# Copyright (c) 2020-2023, NVIDIA CORPORATION
+
+import pandas as pd
+import pytest
+from shapely.geometry import (
+    LineString,
+    MultiLineString,
+    MultiPoint,
+    MultiPolygon,
+    Point,
+    Polygon,
+)
+
+import cuspatial
+
+
+def test_point_intersects_point():
+    g1 = cuspatial.GeoSeries([Point(0.0, 0.0)])
+    g2 = cuspatial.GeoSeries([Point(0.0, 0.0)])
+    gpdg1 = g1.to_geopandas()
+    gpdg2 = g2.to_geopandas()
+    got = g1.intersects(g2)
+    expected = gpdg1.intersects(gpdg2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_points_intersects_points():
+    g1 = cuspatial.GeoSeries([Point(0.0, 0.0), Point(0.0, 0.0)])
+    g2 = cuspatial.GeoSeries([Point(0.0, 0.0), Point(0.0, 1.0)])
+    gpdg1 = g1.to_geopandas()
+    gpdg2 = g2.to_geopandas()
+    got = g1.intersects(g2)
+    expected = gpdg1.intersects(gpdg2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_linestring_intersects_point():
+    g1 = cuspatial.GeoSeries([LineString([(0.0, 0.0), (0.0, 1.0)])])
+    g2 = cuspatial.GeoSeries([Point(0.0, 0.0)])
+    gpdg1 = g1.to_geopandas()
+    gpdg2 = g2.to_geopandas()
+    got = g1.intersects(g2)
+    expected = gpdg1.intersects(gpdg2)
+    pd.testing.assert_series_equal(expected, got.to_pandas())
+
+
+def test_extra_linestring_intersects_point():
```

Comment:
Why are there tests called "extra" and "extra 2", etc? Why not just have multiple test cases? Or just number them? Or more descriptive names?

Reply:
Oh just because the descriptive names can get very long. I could just number these, this pair of tests just tests that the beginning point of the linestring results in correct results, and also that the ending point of the linestring does.